src/model.ts

```diff
@@ -370,15 +370,34 @@ export class Model {
 	}
 }
 
-export interface ModelConstructor {
-	new(createOwnProperties?: boolean): Model;
+export function createModel<TTypes>(options?: ModelTypeOptions<TTypes> & Required<ModelNamespaceOption<TTypes>> & ModelLocalizationOptions, config?: ModelConfiguration): Promise<ModelOfType<TTypes> & ModelWithNamespace<TTypes>>;
+export function createModel<TTypes>(options?: ModelTypeOptions<TTypes> & ModelLocalizationOptions, config?: ModelConfiguration): Promise<ModelOfType<TTypes> & ModelNamespace<TTypes>>;
+export function createModel<TTypes>(options?: ModelOptions<TTypes>, config?: ModelConfiguration): Promise<ModelOfType<TTypes>> {
```

Comment:
_info_: Add a `createModel` function that can be used to provide types for the model and returns a model that knows what types it contains.